src/graphic/Displayable.ts

```diff
@@ -234,6 +235,23 @@ class Displayable<Props extends DisplayableProps = DisplayableProps> extends Ele
             }
         }
 
+        // consider host target
+        if (
```

Comment:
We can treat `__hostTarget` as a special `parent` and doing both check in the

https://github.com/ecomfe/zrender/blob/458d111917e520ac944f6541169990db164f6ccb/src/graphic/Displayable.ts#L229
```ts
if (considerAncestors) {
      let node = this;
      while (node) {
          if (node.ignore) {
              return false;
          }
          node = node.parent || node.__hostTarget;
      }
  }
```

Reply:
Wow, I agree with this way, we don't need to modify the api of `shouldBePainted`, I have fixed this using these codes. Please review again.